elasticai/creator/nn/quantized_grads/fixed_point/_two_complement_fixed_point_config.py

```diff
@@ -0,0 +1,35 @@
+from dataclasses import dataclass
+
+import torch
+
+
+@dataclass
+class FixedPointConfigV2:
```

Comment:
we could add a

```python
class FixedPointConfig:
  __slots__ = ("frac_bits", "total_bits", "stochastic_rounding")
  
  def __init__(self, total_bits: int, frac_bits: int, stochastic_rounding: bool):
    #...
```

This prevents adding more fields to an object of this type after creation and makes attribute access way faster. I.e.,

```python
fxp = FixedPointConfig(4, 2, False)
fxp.something_else = 4 # runtime error
```
Finally, since this lives in the quantized_grads fixed_point package rn, you could just call this `FixedPointConfig` as well, no problem. Another options is: we place it in the same package as the old `FixedPointConfig` and call it `FixedPointConfigV2`.

I think both of these would make it easier to replace the old config by your newer one.

Reply:
I fixed it with making the class immutable. This way I also catch your worries from the other comment.